pkg/skaffold/sync/sync_test.go

```diff
@@ -549,31 +557,41 @@ func TestNewSyncItem(t *testing.T) {
 				Added:    []string{filepath.Join("node", "index.html")},
 				Modified: []string{filepath.Join("node", "server.js")},
 			},
-			dependencies: map[string][]string{"index.html": {"/index.html"}, "server.js": {"/server.js"}},
+			syncRules: []*latest.SyncRule{{Src: "index.html", Dest: "/"}, {Src: "server.js", Dest: "/"}},
 			expected: &Item{
 				Image: "test:123",
 				Copy: map[string][]string{
 					filepath.Join("node", "server.js"):  {"/server.js"},
 					filepath.Join("node", "index.html"): {"/index.html"},
 				},
+				Delete: map[string][]string{},
 			},
 		},
 		{
-			description: "infer: delete not syncable",
+			description: "infer: delete",
 			artifact: &latest.Artifact{
 				Sync: &latest.Sync{
 					Infer: []string{"*"},
 				},
 				Workspace: ".",
 			},
+			builds: []build.Artifact{
+				{
+					Tag: "test:123",
+				},
+			},
 			evt: filemon.Events{
 				Added:   []string{"index.html"},
 				Deleted: []string{"server.html"},
 			},
-			dependencies: map[string][]string{"index.html": {"/index.html"}, "server.html": {"/server.html"}},
-			builds: []build.Artifact{
-				{
-					Tag: "placeholder",
+			syncRules: []*latest.SyncRule{{Src: "index.html", Dest: "/"}, {Src: "server.html", Dest: "/"}},
+			expected: &Item{
+				Image: "test:123",
+				Copy: map[string][]string{
+					filepath.Join("index.html"): {"/index.html"},
```

Comment:
```suggestion
					"index.html": {"/index.html"},
```
same below. Or does that have an effect that I'm not aware of?

Reply:
fixed